File: backend/app/api/schemas/standards.py
"""Schemas for Indian Standards and clause hierarchies."""

import uuid
from typing import List, Optional
from pydantic import BaseModel, Field


class ClauseSummary(BaseModel):
    id: uuid.UUID
    clause_number: str
    heading: Optional[str] = None
    page_start: Optional[int] = None
    page_end: Optional[int] = None
    parent_clause_id: Optional[uuid.UUID] = None


class ClauseDetail(ClauseSummary):
    standard_id: uuid.UUID
    content: str
    children: List[ClauseSummary] = Field(default_factory=list)


class StandardSummary(BaseModel):
    id: uuid.UUID
    standard_number: str
    title: str
    short_title: Optional[str] = None
    edition: Optional[str] = None
    status: str
    publication_date: Optional[str] = None


class StandardDetail(StandardSummary):
    scope: Optional[str] = None
    document_id: Optional[uuid.UUID] = None
    clauses_count: int = 0
    clauses: List[ClauseSummary] = Field(default_factory=list)

File: frontend/types/standards.ts
export interface ClauseSummary {
  id: string;
  clause_number: string;
  heading?: string | null;
  page_start?: number | null;
  page_end?: number | null;
  parent_clause_id?: string | null;
}

export interface ClauseDetail extends ClauseSummary {
  standard_id: string;
  content: string;
  children: ClauseSummary[];
}

export interface StandardSummary {
  id: string;
  standard_number: string;
  title: string;
  short_title?: string | null;
  edition?: string | null;
  status: string;
  publication_date?: string | null;
}

export interface StandardDetail extends StandardSummary {
  scope?: string | null;
  document_id?: string | null;
  clauses_count: number;
  clauses: ClauseSummary[];
}

export interface SearchHit {
  chunk_id: string;
  content: string;
  standard_number?: string | null;
  clause_number?: string | null;
  heading?: string | null;
  pages?: string | null;
  score: number;
  citation_text?: string | null;
}

export interface SearchResponseData {
  query: string;
  normalized_query: string;
  total_results: number;
  duration_ms: number;
  hits: SearchHit[];
}
